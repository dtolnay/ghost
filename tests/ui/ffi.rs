#![feature(generic_const_items)]
#![allow(incomplete_features)]
#![deny(improper_ctypes_definitions)]

use ghost::phantom;

#[phantom]
pub struct MyPhantom<T>;

pub extern "C" fn extern_fn(_phantom: MyPhantom<i32>) {}

fn main() {}
