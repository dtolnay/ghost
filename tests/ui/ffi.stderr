error: `extern` fn uses type `MyPhantom__Phantom<i32>`, which is not FFI-safe
 --> tests/ui/ffi.rs:8:39
  |
8 | pub extern "C" fn extern_fn(_phantom: MyPhantom<i32>) {}
  |                                       ^^^^^^^^^^^^^^ not FFI-safe
  |
  = help: consider adding a `#[repr(C)]`, `#[repr(transparent)]`, or integer `#[repr(...)]` attribute to this enum
  = note: enum has no representation hint
note: the type is defined here
 --> tests/ui/ffi.rs:5:1
  |
5 | #[phantom]
  | ^^^^^^^^^^
note: the lint level is defined here
 --> tests/ui/ffi.rs:1:9
  |
1 | #![deny(improper_ctypes_definitions)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `phantom` (in Nightly builds, run with -Z macro-backtrace for more info)
